protocol/0088-PPRF-party_profile.md

```diff
@@ -0,0 +1,44 @@
+# On-chain Party Profile
+
+The on-chain party profile feature allows users to associate a unique alias and free-form metadata with a public key they own. This data is stored on-chain and is exposed to anyone through APIs.
+
+
+## Updating a party profile
+
+Any user with a public key is able to update their party profile by submitting a signed `UpdatePartyProfile` transaction.
+
+To prevent spam attacks, the number of transactions which can be sent in an epoch per party is restricted and a parties total balance across all assets (expressed in quantum) must exceed `spam.protection.updatePartyProfile.min.funds`. Refer to the [spam protection specification](./0062-SPAM-spam_protection.md#party-profile-spam) for further details on these restrictions.
+
+An `UpdatePartyProfile` has the following fields.
+
+- `alias`: an optional string which can be as an alternative identifier for the public key in any dApp (e.g. in competition leaderboards).
+- `metadata`: an optional list of metadata key value pairs to further describe a public key.
+
+If metadata is specified, the metadata must adhere to the following rules:
+
+- no more than 10 key value pairs can be specified
+- a key must be no longer than 33 characters
+- a value must be no longer than 256 characters
+
+If an alias is specified, it must adhere to the following rules:
+
+- the alias must be no longer than 32 characters
+- the alias must be unique (i.e. the alias must not already be associated with an existing party profile)
+
+**Note:**
+In order to validate uniqueness of an alias, core must keep a record of the aliases associated with each party. If a party is no longer active.
+
+### Alias Squatting
+
+To prevent alias squatting the network will track the number of consecutive epochs each party has funds totalling less than `spam.protection.updatePartyProfile.min.funds` (quantum) across all their accounts.
+
+If the number of consecutive epochs is greater than or equal to `spam.protection.updatePartyProfile.maxInactiveEpochs` the network will no longer associate the alias and metadata with the key.
```

Comment:
I don't think name squatting makes much sense, and should not be implement like so. I could have fund on my party, decide to move it away and come back after that time. If I come back after I exceeded the number of epoch without funds someone can have stollen my profile and usurpate the identity of someone in the network somehow (obviously you can check what alias is attached to what pubkey and all but..).

I think the spam mecanics needs to be good enough so it's difficult to squat a name and it should be enough.

Reply:
This was also to manage storage used, a party could infinitely move funds between keys and update a profile which the network would then have to store.  There would be transfer fees for doing this but ideally `spam.protection.updatePartyProfile.min.funds` is as low as possible so the fees may not be enough of a deterrent to prevent a spam attack.

Reply:
@cdummett - for palazzo we plan to only add spam protection, can we make the name unique and add details regarding spam. 

We can pick up the conversation on name squatting for the nebula milestone